test/test_pebble.py

```diff
@@ -3467,101 +3483,105 @@ def test_wait_output_send_stdin_bytes(self):
             ('TXT', '{"command":"end"}'),
         ]
 
-    def test_wait_output_no_stdout(self):
-        stdio, stderr, _ = self.add_responses('123', 0)
+    def test_wait_output_no_stdout(self, client: MockClient):
+        stdio, stderr, _ = self.add_responses(client, '123', 0)
         stdio.receives.append('{"command":"end"}')
         stderr.receives.append('{"command":"end"}')
         stdout_buffer = io.BytesIO()
-        process = self.client.exec(["echo", "FOOBAR"], stdout=stdout_buffer, encoding=None)
+        process = client.exec(["echo", "FOOBAR"], stdout=stdout_buffer, encoding=None)
         with pytest.raises(TypeError):
             process.wait_output()
 
-    def test_wait_output_bad_command(self):
-        stdio, stderr, _ = self.add_responses('123', 0)
+    def test_wait_output_bad_command(self, caplog: pytest.LogCaptureFixture, client: MockClient):
+        stdio, stderr, _ = self.add_responses(client, '123', 0)
         stdio.receives.append(b'Python 3.8.10\n')
         stdio.receives.append('not json')  # bad JSON should be ignored
         stdio.receives.append('{"command":"foo"}')  # unknown command should be ignored
         stdio.receives.append('{"command":"end"}')
         stderr.receives.append('{"command":"end"}')
 
-        with self.assertLogs('ops.pebble', level='WARNING') as cm:
-            process = self.client.exec(['python3', '--version'])
+        with caplog.at_level(level='WARNING', logger='ops.pebble'):
+            process = client.exec(['python3', '--version'])
             out, err = process.wait_output()
-        assert cm.output == [
-            "WARNING:ops.pebble:Cannot decode I/O command (invalid JSON)",
-            "WARNING:ops.pebble:Invalid I/O command 'foo'",
+        expected = [
+            "Cannot decode I/O command (invalid JSON)",
```

Comment:
Log level/warning is not included in `caplog.records`, so it's removed here, but `caplog.at_level(level='WARNING', logger='ops.pebble')` makes sure it only captures logs with this specific model and level, so the behaviour isn't changed. Same below, there is one more test like this.